nodejs/week1/avg.js

```diff
@@ -0,0 +1,23 @@
+//calculate the average of the numbers provided
+// as command line arguments.
+import { argv } from "process";
+
+let sum = 0;
+const args = argv.slice(2).map((arg) => parseInt(arg));
+console.log(args);
+const res = args.every((element) => typeof element === "number");
+
+//What about if no arguments are provided (node avg.js)?
+if (args.length === 0) {
+  console.log("Please provide number as argument.");
+  process.exit(1);
+}
+//What should happen if the provided arguments are not numbers
+else if (args.filter((num) => !isNaN(num))) {
+  console.log("All arguments must be numbers.");
+  process.exit(1);
+} else {
+  args.forEach((arg) => (sum += arg));
+  const avr = sum / args.length;
+  console.log(avr);
+}
```

Comment:
very nice approach with checking the pessimistic scenarios first! Please notice that you will exit the process in every if statement, so there is no need for nesting - meaning, you don't have to do `else` statements. This makes code more legible in the long run!

Reply:
thankyou for feedback